src/mpyl/project.py

```diff
@@ -134,12 +129,12 @@ class StageSpecificProperty(Generic[T]):
     deploy: Optional[T]
     postdeploy: Optional[T]
 
-    def for_stage(self, stage: Stage) -> Optional[T]:
-        if stage == Stage.BUILD:
+    def for_stage(self, stage: str) -> Optional[T]:
+        if stage == "build":
```

Comment:
We can't have fully dynamic stages if we stick to enums. I'd like to give people the option to define their own stages eventually.

Reply:
I'm just suggesting to replace the current hardcoded strings with an enum, that's not blocking anything or doing anything different than the current solution?

And if we do end up making it fully dynamic with stages coming from the config or something, then it's easier to find and replace the enum (values) compared to hardcoded strings.

Reply:
Yeah, but as a class though (https://github.com/Vandebron/mpyl/pull/255/files#diff-8ab9ddcdb8fee93bb0c9016ade42a4f20cb40bcfa5fd27e03d014c3c5725a542R54-R56) not an enum. 
So we should convert the stage string (from the cli for example) to a `Stage` object ASAP and then pass that around.

Reply:
There are 2 different types though, there is the passed around dynamic stage that comes from the config/cli and (currently) is a string turned into a class. This is the one you are talking about. I'm talking about the strings that are used to say `if passed_around_stage.name == "build": do this specific thing`. In my head we can replace this comparison string "build" with an enum, since it's done multiple times for multiple stages. 

Anyway, it's not a big deal

Reply:
How can stages encoded in an enum be dynamic? We can't introduce a new stage purely through altering the config. If we use the class stage, we can create introduce as many different stages as we want.